components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -387,15 +386,21 @@ private Response handleOAuthProblemException(OAuthMessage oAuthMessage, OAuthPro
             log.debug(e.getError(), e);
         }
 
+        String errorCode = e.getError() != null ? e.getError() : OAuth2ErrorCodes.INVALID_REQUEST;
+        String errorDescription = e.getDescription() != null ? e.getDescription() : e.getMessage();
+        String state = e.getState();
```

Comment:
since this code block is required only if the condition in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1827/files#diff-debebfb0b01b553afd77f73d24e1df466030d3d19b218e0d7b3005364a05939aR394 is not met, shall we move it after this condition?

Reply:
Fixed with e0a7995beb19d81a39eff11df56bc838eac1b687